packages/widget_toolkit_pin/lib/src/lib_pin_code_with_biometrics/blocs/pin_code_bloc.dart

```diff
@@ -169,11 +153,21 @@ class PinCodeBloc extends $PinCodeBloc {
     if (await biometricAuthenticationService.authenticate(localizedReason)) {
       final pinCode = await pinCodeService.getPinCode();
       if (pinCode != null) {
-        return await pinCodeService.verifyPinCode(pinCode);
+        final verificationResult = await pinCodeService.verifyPinCode(pinCode);
+        return verificationResult;
       }
     }
     return false;
   }
 
+  /// Adds a digit to the pin code value
+  Future<String> _addDigit(String digit) async {
```

Comment:
I think we can pick a method name that corresponds to the actual implementation. It seems that we do not add a digit to the pin code but rather compute a new digits based on some criteria. 

WIth that being said would suggest either renaming the method to reflect its implementation (suggestion: `_computeDigits`) or to use a chain of reactive operators in its usage such as

```
_$addDigitEvent
    .asyncMap((digit) async => (digit: digit:, allowedLength: await pinCodeService.getPinLength())))
    .where((data) => _pinCode.value.length <= data.allowedLength)
    .map((data) => _pinCode.value + data.digit)

```

Reply:
Done